compiler/qsc_codegen/src/qir_base.rs

```diff
@@ -64,8 +72,7 @@ pub fn generate_qir(
     );
     match result {
```

Comment:
Not your change, but since you're returning the error directly, wouldn't it simplify the code to do `let val = eval_expr(...)?` above instead of having the `match` here?

Reply:
Same comment for the `generate_qir_for_stmt` change below.

Reply:
Good call!